rancher2/structure_agent_deployment_customization_v2.go

```diff
@@ -0,0 +1,119 @@
+package rancher2
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+// Flatteners
+
+func flattenAgentDeploymentCustomizationV2(in *v1.AgentDeploymentCustomization) []interface{} {
+	obj := make(map[string]interface{})
+	if in == nil {
+		return []interface{}{}
+	}
+
+	if len(in.AppendTolerations) > 0 {
+		obj["append_tolerations"] = in.AppendTolerations
+	}
+
+	if in.OverrideAffinity != nil {
+		obj["override_affinity"] = in.OverrideAffinity
+	}
+
+	if in.OverrideResourceRequirements != nil {
+		obj["override_resource_requirements"] = in.OverrideResourceRequirements
+	}
+
+	return []interface{}{obj}
+}
+
+// Expanders
+
+func expandAgentDeploymentCustomizationOverrideResourceRequirementsV2(p []interface{}) (*corev1.ResourceRequirements, error) {
+	obj := &corev1.ResourceRequirements{}
+	if p == nil || len(p) == 0 || p[0] == nil {
+		return nil, nil
+	}
+
+	for i := range p {
+		in := p[i].(map[string]interface{})
+
+		obj.Limits = corev1.ResourceList{}
```

Comment:
We need to instantiate the map inside any `ResourceList` before we can use it, since the actual type of a `ResourceList` is `map[ResourceName]resource.Quantity`. If we don't I'm pretty sure we will be working with a nil map and line 55 will panic